src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -32,10 +31,22 @@ public static FormulaValue FromJson(string jsonString, FormulaType formulaType =
       
         public static FormulaValue FromJson(string jsonString, FormulaValueJsonSerializerSettings settings, FormulaType formulaType = null)
         {
-            using JsonDocument document = JsonDocument.Parse(jsonString);
-            JsonElement propBag = document.RootElement;
-
-            return FromJson(propBag, settings, formulaType);
+            try
+            {
+                using JsonDocument document = JsonDocument.Parse(jsonString);
+                JsonElement propBag = document.RootElement;
+
+                return FromJson(propBag, settings, formulaType);
+            }
+            catch (JsonException je)
+            {                
+                return new ErrorValue(IRContext.NotInSource(formulaType), new ExpressionError()
+                {
+                    Message = $"{je.GetType().Name} {je.Message} {je.StackTrace}",

```

Comment:
>{je.StackTrace}", [](http://example.com/codeflow?start=65&length=17)

This probably shouldn't be part of the error message. 
ErrorValues get returned to the user. Whereas a C# callstack is not meaningful to an end-user.